### ISSUE TITLE
TLS connections fail on macOS

### ISSUE THREAD

**Issue by strafe** [rcr/rirc#30]
I know this may be hard for you to troubleshoot since I don't believe you use macOS, but it's kind of a bummer and I don't know enough about C/mbedtls to take a look myself.
```
$ ./rirc.debug -s chat.freenode.net

 07:25 ?           ~ Connecting to chat.freenode.net:6697
 07:25 ?           ~  .. Connected [204.225.96.251]
 07:25 ?           ~  .. Establishing TLS connection
 07:25 !           ~  .... The certificate is not correctly signed by the trusted CA
 07:25 !           ~  ....
 07:25 !           ~  .. X509 - Certificate verification failed, e.g. CRL, CA or signature
                   ~ check failed
 07:25 !           ~  .. TLS connection failure
 07:25 !           ~  .. Connection failed -- retrying
 07:25 ?           ~ Attemping reconnect in 00:04
 07:25 ?           ~ Connecting to chat.freenode.net:6697
 07:25 ?           ~  .. Connected [185.30.166.37]
 07:25 ?           ~  .. Establishing TLS connection
 07:25 !           ~  .... The certificate is not correctly signed by the trusted CA
 07:25 !           ~  ....
 07:25 !           ~  .. X509 - Certificate verification failed, e.g. CRL, CA or signature
                   ~ check failed
 07:25 !           ~  .. TLS connection failure
 07:25 !           ~  .. Connection failed -- retrying
 07:25 ?           ~ Attemping reconnect in 00:08
 ```

**Comment by rcr** [comment]
I'm not sure, but it might have something to do with the cert path in macOS. I'll borrow a macbook and look into this.

**Comment by strafe** [comment]
> I'm not sure, but it might have something to do with the cert path in macOS. I'll borrow a macbook and look into this.

Oh perfect! That was it, `CA_CERT_PATH` needed to be `/etc/ssl/`. Maybe there's a clean way to handle that in the default config?

**Comment by rcr** [comment]
I'm not sure if there's a simple automated way to fix this. The cert path could be added to the Makefile with build instructions in the docs to override it?

**Comment by strafe** [comment]
I figured you could make `CA_CERT_PATH` empty in the default config (`""`), and then add some logic in the code that automatically tries to determine the best path based on OS only if it's blank. But whatever you think.

**Comment by rcr** [comment]
That's a good option. Cert loading can try a couple of known paths.

**Comment by strafe** [comment]
Sounds good, here is the tree output for `/etc/ssl/` on macOS if it helps. I noticed that there is a `certs` folder but it's empty.
```
/etc/ssl/
├── cert.pem
├── certs
├── openssl.cnf
└── x509v3.cnf

1 directory, 3 files
```

**Comment by rcr** [comment]
The issue with parsing multiple paths is that `mbedtls_x509_crt_parse_path` has no way of indicating that no certs were loaded.

https://github.com/ARMmbed/mbedtls/issues/3009

**Comment by strafe** [comment]
> The issue with parsing multiple paths is that `mbedtls_x509_crt_parse_path` has no way of indicating that no certs were loaded.
> 
> [ARMmbed/mbedtls#3009](https://github.com/ARMmbed/mbedtls/issues/3009)

Hmm that's unfortunate. I guess it's either wait it out or implement a temporary workaround like checking if `/etc/ssl/certs/` is empty or just detecting the OS at runtime and deciding based on that.

**Comment by rcr** [comment]
This is now fixed in v0.1.5
